File: find-duplicate-O(n)/findDup.js
const findDup = (arr) => {
    
    const dict = {};
    
    for(let i = 0; i < arr.length; i++) {
      if(dict[arr[i]] === 1){
        return arr[i];
      } else {
        dict[arr[i]] = 1;
      };
    };
  };

module.exports = { findDup };

//https://www.codewars.com/kata/558f0553803bc3c4720000af/train/javascript

File: python-venv/find-duplicate-O(n)/findDup.py
def findDup(arr):

    dict = []
    output = 0

    for i in range(len(arr)):
        if arr[i] in dict:
            output = arr[i]
            break
        else:
            dict += [arr[i]]
            
    return output

#https://www.codewars.com/kata/558f0553803bc3c4720000af/train/python